Tella/Data/Database/ServerDatabase.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  UwaziServerDatabase.swift
+//  Tella
+//
+//  Created by Robert Shrestha on 9/14/23.
+//  Copyright © 2023 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+// MARK: - Methods related to Uwazi Server
+extension TellaDataBase {
+    func createServerTable() {
+        let columns = [
+            cddl(D.cServerId, D.integer, primaryKey: true, autoIncrement: true),
+            cddl(D.cName, D.text),
+            cddl(D.cURL, D.text),
+            cddl(D.cUsername, D.text),
+            cddl(D.cPassword, D.text),
+            cddl(D.cAccessToken, D.text),
+            cddl(D.cActivatedMetadata, D.integer),
+            cddl(D.cBackgroundUpload, D.integer),
+            cddl(D.cApiProjectId, D.text),
+            cddl(D.cSlug, D.text),
+            cddl(D.cAutoUpload, D.integer),
+            cddl(D.cAutoDelete, D.integer)
+        ]
+        statementBuilder.createTable(tableName: D.tServer, columns: columns)
+    }
+    
+    func addServer(server : Server)  -> Result<Int, Error> {
+        do {
+            let valuesToAdd = [KeyValue(key: D.cName, value: server.name),
+                               KeyValue(key: D.cURL, value: server.url),
+                               KeyValue(key: D.cUsername, value: server.username),
+                               KeyValue(key: D.cPassword, value: server.password ),
+                               KeyValue(key: D.cAccessToken, value: server.accessToken),
+                               KeyValue(key: D.cActivatedMetadata, value: server.activatedMetadata == false ? 0 : 1),
+                               KeyValue(key: D.cBackgroundUpload, value: server.backgroundUpload == false ? 0 : 1),
+                               KeyValue(key: D.cApiProjectId, value: server.projectId),
+                               KeyValue(key: D.cSlug, value: server.slug),
+                               KeyValue(key: D.cAutoUpload, value:server.autoUpload == false ? 0 : 1),
+                               KeyValue(key: D.cAutoDelete, value:server.autoDelete == false ? 0 : 1)]
+            
+            let serverId = try statementBuilder.insertInto(tableName: D.tServer,
+                                                           keyValue: valuesToAdd)
+            return .success(serverId)
+        } catch let error {
+            debugLog(error)
+            return .failure(error)
+        }
+    }
+
+    func getServers() -> [Server] {
+        var servers: [Server] = []
+        
+        // Function to append servers from a given table
+        func appendServers(fromTable tableName: String, serverType: ServerConnectionType) {
+            do {
+                let serversDict = try statementBuilder.selectQuery(tableName: tableName, andCondition: [])
+                serversDict.forEach { dict in
+                    servers.append(getServer(dictionnary: dict, serverType: serverType))
```

Comment:
I think it's better to create get tella web servers and get Uwazi servers functions separately 

Tella report Server and Uwazi server doesn't have the same table in the database,  so they don't have the same fields

I suggest creating `UwaziServer` and `TellaReportServer` classes that inherit from a Server class containing the shared parameters.

Reply:
Yeah, I agree with this. We should have different classes for every server that we have but changing this will imply a bigger refactor since the class Server is already used in multiple parts of the apps (reports, settings, connections, and so on). What do you think if we left this refactor for the next iteration so we could merge this PR ASAP? I'm worried that it is getting bigger and bigger and we are still working on this. Since there would be no database changes it wouldn't be that big of a deal. That way we can also dedicate more time to test that refactor and think of a better way of how we deal with servers since we would have many more in the future. Let me know if that could work

Reply:
In my opinion it will be more complex in the future if it is not fixed in this feature, I don't think it will take much time and it will be more organized